src/cplscheme/tests/SerialImplicitCouplingSchemeTest.cpp

```diff
@@ -673,17 +702,253 @@ BOOST_AUTO_TEST_CASE(testAccelerationWithLinearExtrapolation)
+    BOOST_TEST(testing::equals(mesh->data(receiveDataIndex)->values()(0), 1.0));
+    // second participant has send data above (should remain untouched)
+    BOOST_TEST(mesh->data(sendDataIndex)->values().size() == 1);
+    BOOST_TEST(testing::equals(mesh->data(sendDataIndex)->values()(0), 4.0));
+  }
+
+  // first window
+  for (int i = 0; i < maxIterations; i++) {
+    // first, second and third iteration
+    BOOST_TEST(cplScheme.isCouplingOngoing());
+
+    if (context.isNamed(first)) {
+      if (i == 0) {
+        // data is uninitialized for first participant
         BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 4);
       } else if (i == 1) {
-        // accelerated data from second participant: 0.5 * 2 + 0.5 * 3 = 2.5
-        BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 2.5);
+        // accelerated data from second participant
+        BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 3); // = 0.5 * 4 + 0.5 * 2
       } else if (i == 2) {
-        // accelerated data from second participant: 0.5 * 2.5 + 0.5 * 3 = 2.75
-        BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 2.75);
+        // accelerated data from second participant
+        BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 2.5); // = 0.5 * 3 + 0.5 * 2
       }
     } else if (context.isNamed(second)) {
-      BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 3); // extrapolation only applied to accelerated data. Second participant receives data from first without acceleration.
+      // data from first participant
+      BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 1);
+    }
+
+    if (i == 0) {
+      BOOST_TEST(cplScheme.isActionRequired(writeIterationCheckpoint));
+      cplScheme.markActionFulfilled(writeIterationCheckpoint);
+      BOOST_TEST(not cplScheme.isActionRequired(readIterationCheckpoint));
+    } else {
+      BOOST_TEST(not cplScheme.isActionRequired(writeIterationCheckpoint));
+      BOOST_TEST(cplScheme.isActionRequired(readIterationCheckpoint));
+      cplScheme.markActionFulfilled(readIterationCheckpoint);
+    }
+
+    // write data to mesh
+    if (context.isNamed(first)) {
+      v << 1.0;
+    } else if (context.isNamed(second)) {
+      v << 2.0;
+    }
+    mesh->data(sendDataIndex)->values() = v;
+    cplScheme.addComputedTime(timestepLength);
+
+    cplScheme.advance();
+
+    if (i < maxIterations - 1) {
+      BOOST_TEST(not cplScheme.isTimeWindowComplete());
+    } else {
+      // window complete since max iterations reached
+      BOOST_TEST(cplScheme.isTimeWindowComplete());
+    }
+  }
+
+  // second window
+  for (int i = 0; i < maxIterations; i++) {
+    // first, second and third iteration
+    BOOST_TEST(cplScheme.isCouplingOngoing());
+    if (context.isNamed(first)) {
+      if (i == 0) {
+        // first order extrapolation uses initial data and final value from last window.
+        BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 0); // = 2*2 - 4
```

Comment:
This part is currently failing, because the extrapolation uses `0` for initial data - even though `4` has been provided above.